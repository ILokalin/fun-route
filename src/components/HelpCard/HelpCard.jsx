import React, { Fragment } from 'react';
import PropTypes from 'prop-types';


export default class extends React.Component {

  static propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    text: PropTypes.array.isRequired
  }

  static defaultProps = {
    id: 'undefined_id',
    title: 'N/A',
    image: '',
    text: ['', '']
  }


  render () {
    const {title, image, text, id} = this.props.helpTopic;

    return (
      <div className="help__card">
        <h3 className="help__title">{title}</h3>
        <div className="help__image-section">
          <img className="help__image" src={image} alt={title} />
        </div>
        <p className="help__text" align="justify" key={id}>
          {text.map((textNode, index) =>  <Fragment key={index}>
                                            {textNode}
                                          </Fragment>
          )}
        </p>
      </div>
    );
  }
}
