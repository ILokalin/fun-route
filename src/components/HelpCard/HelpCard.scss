.help-card {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  width: 100%;
  
  &:hover {
    .help-card__title {
      color: $color-main;
    }
  }


  &__title {
    transition: all 0.3s linear;
    padding-bottom: 3.125rem;
    text-align: center;
  }
  
  
  &__image-section {
    display: flex;
    align-items: center;
    justify-content: center;
    margin-bottom: 2.5rem;
    border-radius: 0.375rem;
    width: 80%;
    transition: background-color 0.8s 0.1s ease;
  
    @media (min-width: $screen-ld) {
      background-color: $color-bg-body;
      height: 14.375rem;
    }
  }
  
  
  &__image {
    border: 1px solid $color-secondary;
    border-radius: 0.25rem;
    transition-duration: 1s;
    transition-timing-function: cubic-bezier(.76, 0, 1, 1);
  }
  
  
  &__text {
    display: block;
    margin: 0 auto;
    padding-bottom: 2.5rem;
    width: 18.75rem;
    line-height: 1.5rem;
    hyphens: manual;
  
    @media (min-width: $screen-md) {
      width: auto;
      margin: 0;
    }
  
    @media (min-width: $screen-sd) {
      max-width: 31.25rem;
    }
    
    &::after {
      display: block;
      clear: both;
      visibility: hidden;
      margin: 0;
      padding: 0;
      height: 0;
      line-height: 0;
      content: "";
    }
  }
  
  
  & .icon {
    transform: translateY(4px);
    margin: 0 0.3125rem;
    width: 1.25rem;
    height: 1.25rem;
  }
  
  
  & .icon__path { 
    fill: $color-secondary;
  }
}
