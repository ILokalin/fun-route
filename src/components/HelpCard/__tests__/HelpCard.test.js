import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import rendrer from 'react-test-renderer';

import HelpCard from 'components/HelpCard';

const fakeHelpTopic = {
  id: 'test_id_0000',
  title: 'test title',
  image: '',
  text: ['test text first', 'test text second']
}

describe('New point field Component create', () => {
  
  it('render snap', () => {
    const component = rendrer.create(<HelpCard 
                                        helpTopic = {fakeHelpTopic}
                                        />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    })
})