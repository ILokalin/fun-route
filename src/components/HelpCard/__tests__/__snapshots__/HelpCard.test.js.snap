// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New point field Component create render snap 1`] = `
<div
  className="help-card"
>
  <h3
    className="help-card__title"
  >
    test title
  </h3>
  <div
    className="help-card__image-section"
  >
    <img
      alt="test title"
      className="help-card__image"
      src=""
    />
  </div>
  <p
    align="justify"
    className="help-card__text"
  >
    test text first
    test text second
  </p>
</div>
`;
